juju.bash

```diff
@@ -83,10 +83,7 @@ function juju::wait
     echo "Waiting for deployment to settle..."
     timeout 45m juju-wait -e "$JUJU_CONTROLLER:$JUJU_MODEL" -w
 
-    ret=$?
-    if (( ret > 0 )); then
-        juju::deploy-failure $ret
-    fi
+    juju::deploy-report $?
```

Comment:
Use deploy-report to short-circuit further testing if `juju-wait` times out or encounters an error

[juju::deploy-report](https://github.com/charmed-kubernetes/jenkins/pull/1138/files#diff-1c740f8e2163fd9b9ae38b82dbe468314d7384fb48606010c0a89d9833c3f024L106-R121)